### PR TITLE
Raise request frontend

### PR DESCRIPTION
**Pull request by shantanu-02**
### Summary

Authenticated users can raise a request for their company logo by providing their Company url.

Issue ticket number : #395 

### Approach

On home page in Try it now section when user searches for a brand logo and doesn't find one an error message is displayed below as Logo Not Available. Beneath this error message a section to Raise a Request will be added with a message Couldn't find your logo try raising a request and a button Raise a Request. Upon clicking this button a modal will open up which is a form asking for user email Id and company URL. Upon this form submission user's request for their logo is processed.

### How to Test the Changes

Go to home page and search for logo. There is no Raise a Request section for user to request their company logo upon failing to find their logo.

### Screenshots or Recordings

TSSS1
TSSS2
TSSS3

### Test coverage report
Screenshot (4)
Screenshot (5)
Screenshot (6)
Screenshot (7)
Screenshot (8)

## Checklist

- [x] I have added/updated tests that cover the changes.
- [ ] I have updated the documentation to reflect the changes.

### CODE REVIEW

**Review by aps08 — CHANGES_REQUESTED** [review]
1. Coverage report missing.
2. msw file/tests missing.

**Comment by Ayushsanjdev** [comment]
Add a coverage report screnshlt in PR of the components file that u worked on and try to make it green.

**Comment by Ayushsanjdev** [comment]
Add a coverage report screnshot in PR of the components file that u worked on and try to make it green.

Also please fix the sonar cloud issue

**Comment by shantanu-02** [comment]
> 1. Coverage report missing.
> 2. msw file/tests missing.

Done with the changes...Please review again
Sonal cloud issue is not fixed beacuse, brand URL's containing HTTP are also being accepted as a request.
Let me know if you want to change this condition @aps08

**Comment by Ayushsanjdev** [comment]
> > 1. Coverage report missing.
> > 2. msw file/tests missing.
> 
> Done with the changes...Please review again
> Sonal cloud issue is not fixed beacuse, brand URL's containing HTTP are also being accepted as a request.
> Let me know if you want to change this condition @aps08 

The coverage that u added is not test coverage, please read about coverage of tests.

**Comment by shantanu-02** [comment]
> > > 1. Coverage report missing.
> > > 2. msw file/tests missing.
> > 
> > 
> > Done with the changes...Please review again
> > Sonal cloud issue is not fixed beacuse, brand URL's containing HTTP are also being accepted as a request.
> > Let me know if you want to change this condition @aps08
> 
> The coverage that u added is not test coverage, please read about coverage of tests.

I have read about it...but how should I generate it?...script from the package.json file is not working for client folder.

**Comment by Ayushsanjdev** [comment]
Like how i told you, try to do like that using coverage command and check if your file is report is good or not, add a screenshot @shantanu-02

File: client/src/components/demo/Demo.test.js [hunk 1]
```diff
@@ -1,59 +1,152 @@
-import {describe, it, expect} from 'vitest';
+import {describe, it, expect, vi} from 'vitest'; // Ensure vi is imported
```
Comment (Ayushsanjdev) [inline client/src/components/demo/Demo.test.js]: Remove the comment, don't blindly copy from chat gpt

**Review by Ayushsanjdev — DISMISSED** [review]
I will review once you add coverage report and a demo video with ui functionality and test cases

**Comment by shantanu-02** [comment]
> Like how i told you, try to do like that using coverage command and check if your file is report is good or not, add a screenshot @shantanu-02

The script in the package.json is configured for Jest, not Vitest, which is why it isn’t working. Should I add the necessary packages and adjust the configuration to generate a coverage report with Vitest? @Ayushsanjdev

**Review by aps08 — APPROVED** [review]

**Review by VinayakaHegade — APPROVED** [review]

**Comment by aps08** [comment]
@VinayakaHegade can you approve this again.

**Review by VinayakaHegade — APPROVED** [review]

**Comment by VinayakaHegade** [comment]
> @VinayakaHegade can you approve this again.

Approved.

**Review by DeltaDynamo — APPROVED** [review]

**Pull request merged**